server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,364 @@
+        "ResourceLimitGroupParser",
+        args -> {
+            Instant currentTimestamp = Instant.now();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd' 'HH:mm:ss", Locale.getDefault())
+                .withZone(ZoneId.of("UTC"));
+            String formattedTimestamp = formatter.format(currentTimestamp);
+            return new ResourceLimitGroup(
+                (String) args[0],
+                UUIDs.randomBase64UUID(),
+                (List<ResourceLimit>) args[1],
+                (String) args[2],
+                formattedTimestamp,
+                formattedTimestamp
+            );
+        }
+    );
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), NAME_FIELD);
+        PARSER.declareObjectArray(
+            ConstructingObjectParser.constructorArg(),
+            (p, c) -> ResourceLimit.fromXContent(p),
+            RESOURCE_LIMITS_FIELD
+        );
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), ENFORCEMENT_FIELD);
+    }
+
+    private static final ConstructingObjectParser<ResourceLimitGroup, Void> PARSER_OPTIONAL_FIELDS = new ConstructingObjectParser<>(
+        "ResourceLimitGroupParser",
+        args -> {
+            Instant currentTimestamp = Instant.now();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd' 'HH:mm:ss", Locale.getDefault())
+                .withZone(ZoneId.of("UTC"));
+            String formattedTimestamp = formatter.format(currentTimestamp);
+            return new ResourceLimitGroup(null, null, (List<ResourceLimit>) args[0], (String) args[1], null, formattedTimestamp);
+        }
+    );
+
+    static {
+        PARSER_OPTIONAL_FIELDS.declareObjectArray(
+            ConstructingObjectParser.optionalConstructorArg(),
+            (p, c) -> ResourceLimit.fromXContent(p),
+            RESOURCE_LIMITS_FIELD
+        );
+        PARSER_OPTIONAL_FIELDS.declareString(ConstructingObjectParser.optionalConstructorArg(), ENFORCEMENT_FIELD);
+    }
+
+    public ResourceLimitGroup(
+        String name,
+        String uuid,
+        List<ResourceLimit> resourceLimits,
+        String enforcement,
+        String createdAt,
+        String updatedAt
+    ) {
+        isValidResourceLimitGroup(name, resourceLimits, enforcement);
+        this.name = name;
+        this.uuid = uuid;
+        this.resourceLimits = resourceLimits;
+        this.enforcement = enforcement;
+        this.createdAt = createdAt;
+        this.updatedAt = updatedAt;
+    }
+
+    public ResourceLimitGroup(StreamInput in) throws IOException {
+        this(in.readString(), in.readString(), in.readList(ResourceLimit::new), in.readString(), in.readString(), in.readString());
+    }
+
+    private void isValidResourceLimitGroup(String name, List<ResourceLimit> resourceLimits, String enforcement) {
```

Comment:
Does this make sense to have the validation logic refactored into their own implementation ?